framework/src/base/NonlinearSystem.C

```diff
@@ -1786,16 +1795,65 @@ NonlinearSystem::computeJacobianInternal(SparseMatrix<Number> &  jacobian)
       {
         _fe_problem.reinitNodeFace(node, boundary_id, 0);
```

Comment:
Yes, this guards against it. I just did not see it in the patch context. So, all is good here as far as I can tell.